Clarify how to restore a database backup in cluster (#3035) (#3098)
neo4j/docs-operations: modules/ROOT/pages/clustering/databases.adoc

[[cluster-seed]]
== Restore a clustered database from a `.backup`/`.dump`

To restore a clustered database from a backup, or to seed a cluster with data:

* You can <<restore-database-using-uri-approach, seed the cluster from a URI>>, where all servers to host the database are seeded with an identical seed from an external source specified by that URI.
* You can use <<cluster-designated-seeder, a _designated seeder_>>, where a designated server is used to create a backed-up database on other servers in the cluster.
* label:new[Introduced in 5.24] You can use <<restore-database-using-recreate-procedure, the `dbms.recreateDatabase()` procedure>>.

[[restore-database-using-uri-approach]]
=== Restore a database from a URI

Before restoring a database from a backup, ensure you have its backup and drop the database on all servers.

> If you have not dropped the database before restoring it, use `CREATE OR REPLACE DATABASE` instead of the `CREATE DATABASE` command.

Author: ```suggestion
```

----

[role=label--new-5.24]
[[restore-database-using-recreate-procedure]]
=== Restore a database using the `dbms.recreateDatabase()` procedure

For detailed information on the recreate procedure, its options and limitations, see xref:database-administration/standard-databases/recreate-database.adoc[].

If you provide a URI to a backup or a dump, the stores on all servers will be replaced by the backup or the dump at the given URI.

> The new database allocations can be put on any `ENABLED` server in the cluster.

Reviewer: Thoughts on re-phrasing this to something like: "The new database can be allocated to any `ENABLED` server in the cluster subject to their mode constraints"? Else it could read as in the user can explicitly choose to put databases on specific servers.

> The new database allocations can be put on any `ENABLED` server in the cluster.

Author: ```suggestion
The new database can be allocated to any `ENABLED` server in the cluster, subject to its mode constraints.
```

You need xref:authentication-authorization/dbms-administration/dbms-database-management-privileges.adoc[the `CREATE DATABASE` and `DROP DATABASE` privileges] to run the recreate procedure.

You do not need to `DROP DATABASE` before calling the `dbms.recreateDatabase()` procedure.

To restore a database using backup as a seed and keeping the original database topology, run the command:

[source, shell]
----
CALL dbms.recreateDatabase("foo", {seedURI: "s3://myBucket/myBackup.backup",});
----

If you need to change the original database topology, run the command:

[source, shell]
----
CALL dbms.recreateDatabase("foo",  {seedURI: "s3://myBucket/myBackup.backup", primaries: 3, secondaries: 0});
----

> Number of newly defined `primaries` and `secondaries` must respect number of `ENABLED` servers in a cluster and their mode constraints.

Reviewer: I think 'The number of..' and 'must respect the...' would read easier.

> Number of newly defined `primaries` and `secondaries` must respect number of `ENABLED` servers in a cluster and their mode constraints.

Author: ```suggestion
The number of newly defined `primaries` and `secondaries` must respect the number of `ENABLED` servers in a cluster and their mode constraints.
```
